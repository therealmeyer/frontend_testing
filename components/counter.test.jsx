import React from 'react';
import { shallow } from 'enzyme';
import Counter from './counter';
import Foo from './foo';
import renderer from 'react-test-renderer';

describe('Counter', () => {
  it("starts with a count of 0", () => {
    let container = shallow(<Counter />);
    let counter = container.state().count;
    expect(counter).toBe(0);
  });
  it("renders 'count' foos", () => {
    let container = shallow(<Counter />);
    expect(container.find("Foo").length).toBe(0);
    container.find("button.add1").simulate("click");
    expect(container.find("Foo").length).toBe(1);

    container.find("button.add1").simulate("click");
    container.find("button.add1").simulate("click");
    container.find("button.add1").simulate("click");

    expect(container.find("Foo").length).toBe(4);
  });
  it('matches the initial snapshot', () => {
    const component = renderer.create(<Counter />)
    expect(component.toJSON()).toMatchSnapshot();

    component.getInstance().add1();
    component.getInstance().add1();

    expect(component.toJSON()).toMatchSnapshot();

  })
})