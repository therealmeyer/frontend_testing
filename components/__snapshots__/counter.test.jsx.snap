// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter matches the initial snapshot 1`] = `
<div>
  <p>
    count: 
    0
  </p>
  <button
    className="add1"
    onClick={[Function]}
  />
</div>
`;

exports[`Counter matches the initial snapshot 2`] = `
<div>
  <p>
    count: 
    2
  </p>
  <button
    className="add1"
    onClick={[Function]}
  />
  <h1>
    Foo
  </h1>
  <h1>
    Foo
  </h1>
</div>
`;
